test/integration/execution/transaction.js

```diff
@@ -681,9 +676,9 @@ module.exports = function (knex) {
       const mysqlKillConnection = async (connection) =>
         knex.raw('KILL ?', [connection.threadId]);
       const killConnectionMap = {
-        mysql: mysqlKillConnection,
-        mysql2: mysqlKillConnection,
-        pg: async (connection) =>
+        [drivers.MySQL]: mysqlKillConnection,
```

Comment:
Example of using the constants in here to ensure the `driverName` matches. May be overkill, but it is an experiment to gauge interest.

Reply:
This does look like a great idea!
Will you add separate branch for pg-native here later on? Wonder if we can somehow cover all multi-dialect dbs without multiple entries.

Reply:
I've thought about it a bit this afternoon. The idea would also be that this could be used in places like: 
https://github.com/knex/knex/pull/4356/files#diff-f275a3d5b4fde0fe972896d099c1eac3ae2614ff845e49a0f73e276ee925e8b5R985 to ensure it always points toward a valid type. This may be overkill because I think some of these areas check to see if it's a valid strategy. 

As for answering the direct question:

In the situation of a map where the types are implicitly matched, duplicating seems like the only option. Switch statements will offer roughly the same ability. I found utilizing `else if` style, as in some areas like this: https://github.com/knex/knex/pull/4356/files#diff-f275a3d5b4fde0fe972896d099c1eac3ae2614ff845e49a0f73e276ee925e8b5R1482 worked well so that the same helpers could be used without much structure change. This lets `isPgBased` be used for `pg`, `pg-native`, and `pg-redshift`, `isMySQL` be used for those grouping, etc. It's not the most elegant, but it does allow the encapsulation to be maintained.

There are definitely other options, but I'm curious of your thoughts. Additionally we can do nothing further with this PR (or even back them out) and come back to it after some more thought goes into it.

Reply:
Given the number of occurrences of the references in the tests, doing a manual conversion any further may not yield the best ROI (given risk and time allocations).